overlord/snapstate/snapstate.go

```diff
@@ -1358,6 +1358,102 @@ func installWithDeviceContext(st *state.State, name string, opts *RevisionOption
 	return doInstall(st, &snapst, snapsup, 0, fromChange, nil)
 }
 
+func Download(ctx context.Context, st *state.State, name string, opts *RevisionOptions, userID int, flags Flags, deviceCtx DeviceContext) (*state.TaskSet, error) {
```

Comment:
Why have this wrapper?

Reply:
I wonder if for our use case we will need to give this a (optional) path and teach download-snap to write there instead of the default place?

Reply:
That makes sense. For our use case, do we not want the downloaded snaps to live with the rest of the snaps that are downloaded on the system?

Reply:
If we don't change the location that we download the snaps to, then I think we can limit the modifications we make later by reusing this section of the code: https://github.com/snapcore/snapd/blob/master/overlord/devicestate/handlers_systems.go?plain=1#L158-L179.

However, I can see how it could be confusing to have snaps that are not installed downloaded into the same directory.

Reply:
I've updated `Download` to take in a path that can be used to specify a directory where the snap will be downloaded to.